code/headers/gga.hpp

```diff
@@ -11,18 +22,81 @@ namespace r2d2::location {
      *
      */
     struct gga_s {
+        // UTC time of fix
         uint32_t time;
-        float latitude;
-        char north_south_hemisphere;
-        float longitude;
-        char east_west_hemisphere;
-        uint8_t fix_quality;
-        uint8_t satellites_tracked;
+
+        // Latitude/ longitude
+        degrees latitude;
+        degrees longitude;
+
+        // Horizontal dilution of precision. The lower the number the better.
+        // This number is lower if the satellites are more spread out.
+        // https://the5krunner.com/2018/07/18/gps-accuracy-hdop-pdop-gdop-multipath-the-atmosphere/
         float horizontal_dilution;
+
+        // Altitude above sea level. The precision of the altitude is always
+        // lower than the lat/long. The precision of the altitude is not in the
+        // gga message.
         float altitude;
-        char altitude_measurement;
+
+        // altitude using the WGS-84 ellipsoid
         float geoid_height;
-        char geoid_height_measurement;
+
+        // the age in seconds since the last update from the diff reference
+        // station.
+        uint32_t diff_age;
+
+        // the id of the diff reference station
+        uint16_t diff_id;
+
+        // variable for how reliable the data is.
+        // 0 = invalid
+        // 1 = GPS fix (SPS)
+        // 2 = DGPS fix, OmniSTAR VBS
+        // 3 = PPS fix
+        // 4 = Real Time Kinematic, fixed integers
+        // 5 = Float RTK, float integers, OmniSTAR XP/HP or Location RTK
+        // 6 = estimated (dead reckoning) (2.3 feature)
+        // 7 = Manual input mode
+        // 8 = Simulation mode
+        uint8_t fix_quality;
```

Comment:
Wouldn't this be the perfect case for an `enum class`?
considering you have 8 distinct values with a meaning.